src/uu/rm/src/rm.rs

```diff
@@ -326,6 +326,24 @@ pub fn remove(files: &[&OsStr], options: &Options) -> bool {
 fn handle_dir(path: &Path, options: &Options) -> bool {
     let mut had_err = false;
 
+    let del = std::path::MAIN_SEPARATOR;
+
+    if path
+        .to_str()
+        .unwrap()
+        .ends_with(format!("{}.", del).as_str())
+        || path
+            .to_str()
+            .unwrap()
```

Comment:
1. The conversion is done twice. That's not a big deal, but maybe just saving the result in a variable is easy enough to avoid doing the same work twice?
2. `.to_str().unwrap()` will cause problems with non-UTF-8 filenames. Please consider using os_str_as_bytes: If it works great (and add a test), if it doesn't (or is too much work or feels out of scope) then disregard this suggestion.
3. I'm not sure whether this is the right condition. Maybe a trailing path separator ("del") should count, too? Here's a small example that demonstrates the difference:
    ```
    $ rm -rf foo && mkdir -p foo/bar/baz && rm -rf foo/bar/baz/../
    rm: refusing to remove '.' or '..' directory: skipping 'foo/bar/baz/../'
    [$? = 1]
    $ find foo/
    foo/
    foo/bar
    foo/bar/baz
    $ rm -rf foo && mkdir -p foo/bar/baz && cargo run -q --features rm rm -rf foo/bar/baz/../
    rm: cannot remove 'foo/bar/baz/../': No such file or directory (os error 2)
    [$? = 1]
    $ find foo
    foo
    foo/bar
    ```
    As you can see, we still delete a directory, causing a confusing error message. I'm not sure whether it suffices to add the two conditions `ends_with("/./")` and ".ends_with("/../")". Can you find any other edge cases?